### PR TITLE
[patching] Adapt aaf recipe to patching format in conandata

### PR DESCRIPTION
**Pull request by danimtb**
Using the new tool helper and adding more information to the conandata file

Example of output:
```
aaf/1.2.0: Apply patch (portability): Simpler CMakeLists.txt
aaf/1.2.0: Apply patch: Link CoreFoundation and CoreServices with find_library
aaf/1.2.0: Apply patch (conan)
```

### CODE REVIEW

File: recipes/aaf/all/conandata.yml [hunk 1]
```diff
@@ -6,11 +6,17 @@ patches:
   "1.2.0":
     - patch_file: "patches/1.2.0-001-simpler-cmakelists.patch"
       base_path: "source_subfolder"
+      patch_type: "portability"
+      patch_source: "https://github.com/google/flatbuffers/pull/5650"
```
Comment (MartinDelille) [inline recipes/aaf/all/conandata.yml:10]: As the author of the first version of this PR I'm interested in these improvements! Do we agree that the `patch_source` is wrong here ?

**Review by MartinDelille — CHANGES_REQUESTED** [review]

File: recipes/aaf/all/conandata.yml (hunk 1 above)
Comment (danimtb) [reply recipes/aaf/all/conandata.yml:10]: Hi @MartinDelille! thanks for commenting 😄 yes, it is wrong. I have just opened the PR to see the reaction of the CI. Also patch type might not match. I will be fixing all those before merging of course. If you have suggestions for the value of the new fields, feel free to comment in the review of the PR. Thanks a lot! 😀

Comment (MartinDelille) [reply recipes/aaf/all/conandata.yml:10]: This is a great improvement for project on Github! This one is on sourceforge though: http://aaf.sourceforge.net/

**Comment by danimtb** [comment]
@MartinDelille I have updated the patches details with the real values for this recipe. I am mostly missing the patch source for each of them (not mandatory) as I am not aware of its origin (I think most of them were created by yourself). Also, I did my best to assign a correct patch type but maybe I am wrong in some of them (here you can see the documentation https://github.com/jgsogo/conan-center-index/pull/5/files#diff-0f4e28a529147ed74e8a4845a4d74179e15c0ebf519c427d51f72324972e450e). Your help here will be appreciated 😃

**Review by prince-chrismc — APPROVED** [review]
🚀 Nice to see this feature coming to life

**Review by uilianries — APPROVED** [review]

**Pull request merged**